example/image-classification/train_cifar10.py

```diff
@@ -36,13 +36,28 @@
 # download data if necessary
 def _download(data_dir):
     if not os.path.isdir(data_dir):
-        os.system("mkdir " + data_dir)
+        os.makedirs(data_dir)  # Recursive directory creation
     os.chdir(data_dir)
     if (not os.path.exists('train.rec')) or \
        (not os.path.exists('test.rec')) :
-        os.system("wget http://webdocs.cs.ualberta.ca/~bx3/data/cifar10.zip")
-        os.system("unzip -u cifar10.zip")
-        os.system("mv cifar/* .; rm -rf cifar; rm cifar10.zip")
+        url = 'http://webdocs.cs.ualberta.ca/~bx3/data/cifar10.zip'
+        if sys.version_info.major == 2:
+            from urllib import urlretrieve
+        elif sys.version_info.major == 3:
+            from urllib.request import urlretrieve
+        urlretrieve(url, 'cifar10.zip')
+        import zipfile
```

Comment:
Yes, zipfile is in python standard library!

At 2016-06-04 02:43:11, "Eric Junyuan Xie" <notifications@github.com> wrote:


In example/image-classification/train_cifar10.py:

>      os.chdir(data_dir)
>      if (not os.path.exists('train.rec')) or \
>         (not os.path.exists('test.rec')) :
> -        os.system("wget http://webdocs.cs.ualberta.ca/~bx3/data/cifar10.zip")
> -        os.system("unzip -u cifar10.zip")
> -        os.system("mv cifar/* .; rm -rf cifar; rm cifar10.zip")
> +        url = 'http://webdocs.cs.ualberta.ca/~bx3/data/cifar10.zip'
> +        if sys.version_info.major == 2:
> +            from urllib import urlretrieve
> +        elif sys.version_info.major == 3:
> +            from urllib.request import urlretrieve
> +        urlretrieve(url, 'cifar10.zip')
> +        import zipfile


is zipfile available by default?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.